Add tests for validating alt_aliases of a canonical alias event.

clokep opened the pull request:
This adds tests for the behavior for validating the `alt_aliases` field of a canonical alias event added to Synapse in matrix-org/synapse#6971.

clokep commented on tests/30rooms/05aliases.pl:
```diff
@@ -99,6 +100,64 @@
+               alt_aliases => [ $bad_alias ],
+            }
+         )->main::expect_http_4xx
+            ->SyTest::pass_on_done( "m.room.canonical_alias rejects missing aliases" );
```
I wonder if these should assert the resulting content to ensure the proper error code (and HTTP code) is returned?

richvdh replied:
wouldn't hurt. the incantation you're after is `main::expect_matrix_error`.

clokep commented:
Good thing I added those -- I realized one of the tests was hitting a different condition than I thought. I added an additional test, so I'm going to request review again.
